flake: TestWatchWorkspaceAgentDevcontainers/PayloadTooLarge

Seen [here](https://github.com/coder/coder/actions/runs/16993743728/job/48179754150)
```
2025-08-15T16:05:17.2177082Z     --- FAIL: TestWatchWorkspaceAgentDevcontainers/PayloadTooLarge (10.19s)
```

--- a/coderd/workspaceagents_test.go
+++ b/coderd/workspaceagents_test.go
@@ -1579,12 +1579,12 @@ func TestWatchWorkspaceAgentDevcontainers(t *testing.T) {
 		t.Parallel()
 
 		var (
-			ctx    = testutil.Context(t, testutil.WaitLong)
-			logger = slogtest.Make(t, &slogtest.Options{IgnoreErrors: true}).Leveled(slog.LevelDebug)
+			ctx               = testutil.Context(t, testutil.WaitLong)
+			logger            = slogtest.Make(t, &slogtest.Options{IgnoreErrors: true}).Leveled(slog.LevelDebug)
 			mClock            = quartz.NewMock(t)
 			updaterTickerTrap = mClock.Trap().TickerFunc("updaterLoop")
-			mCtrl  = gomock.NewController(t)
-			mCCLI  = acmock.NewMockContainerCLI(mCtrl)
+			mCtrl             = gomock.NewController(t)
+			mCCLI             = acmock.NewMockContainerCLI(mCtrl)
 
 			client, db = coderdtest.NewWithDatabase(t, &coderdtest.Options{Logger: &logger})
 			user       = coderdtest.CreateFirstUser(t, client)

--- a/coderd/workspaceagents_test.go
+++ b/coderd/workspaceagents_test.go
@@ -1579,8 +1579,10 @@ func TestWatchWorkspaceAgentDevcontainers(t *testing.T) {
 		t.Parallel()
 
 		var (
-			ctx    = testutil.Context(t, testutil.WaitShort)
+			ctx    = testutil.Context(t, testutil.WaitLong)
 			logger = slogtest.Make(t, &slogtest.Options{IgnoreErrors: true}).Leveled(slog.LevelDebug)
+			mClock            = quartz.NewMock(t)
+			updaterTickerTrap = mClock.Trap().TickerFunc("updaterLoop")
 			mCtrl  = gomock.NewController(t)
 			mCCLI  = acmock.NewMockContainerCLI(mCtrl)
 
@@ -1621,6 +1623,7 @@ func TestWatchWorkspaceAgentDevcontainers(t *testing.T) {
 			o.Logger = logger.Named("agent")
 			o.Devcontainers = true
 			o.DevcontainerAPIOptions = []agentcontainers.Option{
+				agentcontainers.WithClock(mClock),
 				agentcontainers.WithContainerCLI(mCCLI),
 				agentcontainers.WithWatcher(watcher.NewNoop()),
 			}
@@ -1630,6 +1633,9 @@ func TestWatchWorkspaceAgentDevcontainers(t *testing.T) {
 		require.Len(t, resources, 1, "expected one resource")
 		require.Len(t, resources[0].Agents, 1, "expected one agent")
 		agentID := resources[0].Agents[0].ID
+
+		updaterTickerTrap.MustWait(ctx).MustRelease(ctx)
+		defer updaterTickerTrap.Close()
 
 		containers, closer, err := client.WatchWorkspaceAgentContainers(ctx, agentID)
 		require.NoError(t, err)

